crates/services/compression/src/ports/configuration.rs

```diff
@@ -0,0 +1,7 @@
+use crate::config::CompressionConfig;
+
+/// Configuration for the compression service
+pub trait CompressionConfigProvider {
+    /// getter for the compression config
+    fn config(&self) -> CompressionConfig;
```

Comment:
From my perspective of ports and adapter, on each module we define ports for interactions that we need in order to make this module works, we don't define ports for exposing this modules functionnalities, this is done when implementing the adapters through the port traits of modules that actually wants it.

Reply:
sure, but we decouple the configuration here ~ in `fuel-core`, we can just implement
`impl CompressionConfigProvider for Config {}`

tbh, we could just use an `Into<CompressionConfig>` trait bound, but this seems cleaner. we do the same thing for gas price service i thinkso

Reply:
the former :)

Reply:
Okey then i'm fine.